### DISCUSSION TITLE
Climate modes should ignore case when setting preset mode, fan mode and swing mode

### DISCUSSION THREAD

**Discussion by brianegge** [home-assistant/architecture#1039] [Entity Models]
Two climate integrations may support the same modes but use different case, such as the Generic Thermostat and Ecobee thermostat. As a human you are left to guess which they use, and if you attempt to set all your thermostats to the same mode, some will fail (and again you get to guess which one failed). 

i.e.,
![image](https://github.com/home-assistant/architecture/assets/175930/9b0ad070-95da-441c-b0a5-3020793dd224)

![image](https://github.com/home-assistant/architecture/assets/175930/d59ce739-d3ac-4873-aa87-03669a83bcda)

(https://community.home-assistant.io/t/hvac-mode-case-sensitive/689394/3)

The component knows which modes it accepts and can easily convert to the correct case for each integration. I.e., https://github.com/home-assistant/core/pull/110548/commits/f54796802f5b52d9839aeee6bc10bf1715e06955

As a user, I want to set all my thermostats to Away when I leave. Today, I have to split that service call into two, one for `Away` and the other for `away`, and only through guessing do I know what thermostat must be in each group.

**Comment by gjohansson-ST** [comment]
I think the correct solution here is to raise and issue to whichever integration is implementing `Away` to change to into `away` which is part of the default implementation of the `climate` entity.
Also the error message could contain the `entity_id` to stop guessing which would again be another issue to raise

Doing the conversions you mention we shouldn't do while I get why you say this is annoying.
I would suggest we close this and you raise the proper issues 👍

**Comment by brianegge** [comment]
For the Ecobee integration, the user can define modes, though you can't define a lowercase mode along side the uppercase one. I.e.,

![56BC6541-4070-4988-8169-6C0A95E59083_1_201_a](https://github.com/home-assistant/architecture/assets/175930/f943da51-0516-413e-8156-fe6cddadda65)

![B05F6F08-0B83-47F2-BD7F-516A9D222F0E_1_201_a](https://github.com/home-assistant/architecture/assets/175930/1debcbe0-b937-4da5-a658-7cadc7aabdf1)

And delightfully, these become presets in the GUI:
![B8C39454-69BF-45AF-92EB-5C2C829AA0F2_1_102_o](https://github.com/home-assistant/architecture/assets/175930/d09a527d-845f-40b5-b0d9-655163b89a6e)

So, should the Ecobee integration search to see if any of the modes match a standard mode, and convert those back and forth? Because the mode validation happens in the base entity (and is final), a given integration doesn't have the opportunity to normalize an input to match it's own preset modes. It would need to look at the presets the device has, see which ones are close matches in the base entity, then create a two way mapping for just those which differ.

**Reply by gjohansson-ST** [reply]
What you're describing is an implementation detail explicitly in the ecobee integration.
the implementation of ecobee could have solved that with a prefix, suffix or whatever to distinguish between the ecobee ones and possible built-ins that one would use.

I would suggest you instead put in a feature request in the forum to get ecobee to lowercase these as that seems to be the issue for you.

**Reply by brianegge** [reply]
It is something which can be fixed generically which improves the user experience.
